pkg/kubectl/cmd/util/openapi/openapi.go

```diff
@@ -266,96 +292,95 @@ func (o *Resources) isPrimitive(s spec.Schema) bool {
+	if len(s.GetProperties().GetAdditionalProperties()) > 0 {
 		// Open API can have embedded type definitions, but Kubernetes doesn't generate these.
 		// This should just be a sanity check against changing the format.
 		return false
 	}
-	if len(s.Type) > 0 {
+	if len(s.GetType().GetValue()) > 0 {
 		// Definition references won't have a type
 		return false
 	}
 
-	p := s.SchemaProps.Ref.GetPointer().String()
-	return len(p) > 0 && strings.HasPrefix(p, "/definitions/")
+	p := s.GetXRef()
+	return len(p) > 0 && strings.HasPrefix(p, "#/definitions/")
 }
 
 // getElementType returns the type of an element for arrays
 // returns an error if s is not an array.
-func (o *Resources) getElementType(s spec.Schema) (spec.Schema, error) {
+func (o *Resources) getElementType(s *openapi_v2.Schema) (*openapi_v2.Schema, error) {
 	if !o.isArray(s) {
-		return spec.Schema{}, fmt.Errorf("%v is not an array type", s.Type)
+		return &openapi_v2.Schema{}, fmt.Errorf("%v is not an array type", o.getTypeNameForField(s))
 	}
-	return *s.Items.Schema, nil
+	return s.GetItems().GetSchema()[0], nil
 }
 
-// getElementType returns the type of an element for maps
+// getValueType returns the type of an element for maps
 // returns an error if s is not a map.
-func (o *Resources) getValueType(s spec.Schema) (spec.Schema, error) {
+func (o *Resources) getValueType(s *openapi_v2.Schema) (*openapi_v2.Schema, error) {
 	if !o.isMap(s) {
-		return spec.Schema{}, fmt.Errorf("%v is not an map type", s.Type)
+		return &openapi_v2.Schema{}, fmt.Errorf("%v is not an map type", o.getTypeNameForField(s))
 	}
-	return *s.AdditionalProperties.Schema, nil
+	return s.GetAdditionalProperties().GetSchema(), nil
 }
 
 // nameForDefinitionField returns the definition name for the schema (field) if it is a complex type
-func (o *Resources) nameForDefinitionField(s spec.Schema) string {
-	p := s.SchemaProps.Ref.GetPointer().String()
+func (o *Resources) nameForDefinitionField(s *openapi_v2.Schema) string {
+	p := s.GetXRef()
 	if len(p) == 0 {
 		return ""
 	}
 
 	// Strip the "definitions/" pieces of the reference
-	return strings.Replace(p, "/definitions/", "", -1)
+	return strings.Replace(p, "#/definitions/", "", -1)
 }
 
 // getGroupVersionKind implements OpenAPIData
 // getGVK parses the gropuversionkind for a resource definition from the x-kubernetes
 // extensions
-// Expected format for s.Extensions: map[string][]map[string]string
 // map[x-kubernetes-group-version-kind:[map[Group:authentication.k8s.io Version:v1 Kind:TokenReview]]]
-func (o *Resources) getGroupVersionKind(s spec.Schema) (schema.GroupVersionKind, error) {
+func (o *Resources) getGroupVersionKind(s *openapi_v2.Schema) (schema.GroupVersionKind, error) {
 	empty := schema.GroupVersionKind{}
 
 	// Get the extensions
-	extList, f := s.Extensions[groupVersionKindExtensionKey]
+	extList, f := vendorExtensionToMap(s.GetVendorExtension())[groupVersionKindExtensionKey]
```

Comment:
Maybe save the result of `vendorExtensionToMap(s.GetVendorExtension())` to avoid process this again below.